app/src/renderer/components/governance/PageProposal.vue

```diff
@@ -0,0 +1,174 @@
+<template lang="pug">
+tm-page(data-title='Proposal')
+  template(slot="menu-body"): tm-balance
+  div(slot="menu"): tm-tool-bar
+    router-link(to="/governance" exact): i.material-icons arrow_back
+
+  tm-data-error(v-if="!proposal")
+
+  template(v-else)
+    .validator-profile__header.validator-profile__section.proposal
+      .column.validator-profile__header__info
+        .row.validator-profile__header__name
+          .top.column
+            div.validator-profile__status-and-title
+              span.validator-profile__status(v-bind:class="status.color" v-tooltip.top="status.message")
+              .validator-profile__header__name__title {{ proposal.title }}
+          .column.validator-profile__header__actions
+            tm-btn(v-if="status.button === 'vote'" value="Vote" color="primary")
+            tm-btn(v-if="status.button === 'deposit'" value="Deposit" color="primary")
+            tm-btn(v-if="!status.button" disabled value="Deposit / Vote" color="primary")
+
+        .row.description
+          p This {{ proposalType }} proposal ({{ `#` + proposal.proposal_id }}) was submitted at block {{ submitBlock }} and voting started at {{ voteBlock }}.
+
+        .row.validator-profile__header__data.votes
+          dl.colored_dl
+            dt Deposit
+            dd {{ proposal.total_deposit[0].amount + ` ` + proposal.total_deposit[0].denom }}
+          .validator-profile__header__data__break
+          dl.colored_dl
+            dt Yes
+            dd {{ proposal.tally_result.yes }} / {{ yesPercentage }}
+          dl.colored_dl
+            dt No
+            dd {{ proposal.tally_result.no }} / {{ noPercentage }}
+          dl.colored_dl
+            dt No with Veto
+            dd {{ proposal.tally_result.no_with_veto }} / {{ noWithVetoPercentage }}
+          dl.colored_dl
+            dt Abstain
+            dd {{ proposal.tally_result.abstain }} / {{ abstainPercentage }}
+
+    .validator-profile__details.validator-profile__section
+      .column
+        .row
+          text-block(:content="proposal.description")
+</template>
+
+<script>
+import num from "scripts/num"
+import { TmBtn, TmPage, TmToolBar } from "@tendermint/ui"
+import TmBalance from "common/TmBalance"
+import FieldVote from "common/TmFieldVote"
+import TextBlock from "common/TextBlock"
+export default {
+  name: `page-proposal`,
+  props: [`proposal`],
+  components: {
+    TmBalance,
+    TmBtn,
+    FieldVote,
+    TmToolBar,
+    TmPage,
+    TextBlock
+  },
+  computed: {
+    proposalType() {
+      return this.proposal.proposal_type.toLowerCase()
+    },
+    submitBlock() {
+      return `#` + num.prettyInt(this.proposal.submit_block)
+    },
+    voteBlock() {
+      if (this.proposal.submit_block === this.proposal.voting_start_block) {
+        return `the same block`
+      } else {
+        return `block #` + num.prettyInt(this.proposal.voting_start_block)
+      }
+    },
+    status() {
```

Comment:
@NodeGuy @faboweb 

here is the switch. what would you recommend instead?

also here is a screenshot of codecov saying the 4 break lines are untested:
<img width="675" alt="screen shot 2018-11-08 at 1 01 32 pm" src="https://user-images.githubusercontent.com/6021933/48217839-784ffa00-e356-11e8-8af8-4dc394fcb6e6.png">